service/vehicle.go

```diff
@@ -0,0 +1,84 @@
+package service
+
+import (
+	"encoding/csv"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/edgarduran/Go-Dispatch-Bootcamp/models"
+)
+
+func GetAllVehicles() ([]models.Vehicle, error) {
+	f, err := os.Open("vehicles.csv")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	data, err := csvReader.ReadAll()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("var1 = %T\n", data)
+	var vehicleList []models.Vehicle
+	for _, vehicle := range data {
+		var rec models.Vehicle
+		for j, field := range vehicle {
+			if j == 0 {
+				id, _ := strconv.ParseInt(field, 10, 64)
+				rec.Id = id
+			} else if j == 1 {
+				rec.Make = field
+			} else if j == 2 {
+				rec.Model = field
+			}
+		}
+
+		vehicleList = append(vehicleList, rec)
+	}
```

Comment:
I take the liberty to improve your code a little bit, with some useful tips:

- Move the `var rec models.Vehicles` line outside of the for loop; the way you had it you are creating a new instance in every iteration of the loop.
- Remove the inner loop; no need to use a for cycle for these elements, and also the extra effort to know which element it was.
- You should never omit errors; by default, a parse error will set the id to 0 (the default value of int64), and maybe you don't want that.

```suggestion
	var vehicleList []models.Vehicle
	var rec models.Vehicle

	for _, vehicle := range data {
		id, err := strconv.ParseInt(vehicle[0], 10, 64)
		if err != nil {
			log.Printf("Unable to parse %s to int, a -1 value will be used instead", vehicle[0])
			id = -1
		}
		
		rec.ID = id
		rec.Make = vehicle[1]
		rec.Model = vehicle[2]

		vehicleList = append(vehicleList, rec)
	}
```